test/integration/testpaths/testpath_reduce_vm_deployments.py

```diff
@@ -0,0 +1,2073 @@
+
+        volumes = list_volumes(
+            self.apiclient,
+            virtualmachineid=self.virtual_machine_with_disk.id,
+            type='ROOT',
+            listall=True
+        )
+
+        snapshot = Snapshot.create(
+            self.apiclient,
+            volumes[0].id,
+            account=self.account.name,
+            domainid=self.account.domainid
+        )
+        self.debug("Snapshot created: ID - %s" % snapshot.id)
+
+        snapshots = list_snapshots(
+            self.apiclient,
+            id=snapshot.id
+        )
+        self.assertEqual(
+            isinstance(snapshots, list),
+            True,
+            "Check list response returns a valid list"
+        )
+
+        self.assertNotEqual(
+            snapshots,
+            None,
+            "Check if result exists in list item call"
+        )
+        self.assertEqual(
+            snapshots[0].id,
+            snapshot.id,
+            "Check resource id in list resources call"
+        )
+
+        self.assertIsNotNone(snapshots[0].zoneid,
+                             "Zone id is not none in listSnapshots")
+        self.assertEqual(
+            snapshots[0].zoneid,
+            self.zone.id,
+            "Check zone id in the list snapshots"
+        )
+
+        self.debug(
+            "select backup_snap_id, account_id, volume_id from "
+            "snapshots where uuid = '%s';" % str(snapshot.id))
+        qresultset = self.dbclient.execute(
+            "select backup_snap_id, account_id, volume_id"
+            " from snapshots where uuid = '%s';" % str(snapshot.id))
+        self.assertNotEqual(
+            len(qresultset),
+            0,
+            "Check DB Query result set"
+        )
+
+        qresult = qresultset[0]
+
+        snapshot_uuid = qresult[0]      # backup_snap_id = snapshot UUID
+
+        self.assertNotEqual(
+            str(snapshot_uuid),
+            'NULL',
+            "Check if backup_snap_id is not null"
+        )
+
+        self.assertTrue(
+            is_snapshot_on_nfs(
+                self.apiclient,
+                self.dbclient,
+                self.config,
+                self.zone.id,
+                snapshot.id))
+        return
+
+    @attr(
+        tags=[
+            "advanced",
+            "advancedns",
+            "smoke",
+            "basic",
+            "eip",
+            "sg"],
+        required_hardware="false")
+    def test_02_edit_service_offering(self):
+        """Test to update existing service offering"""
+
+        # Validate the following:
+        # 1. updateServiceOffering should return
+        #    a valid information for newly created offering
+
+        # Generate new name & displaytext from random data
+        random_displaytext = random_gen()
+        random_name = random_gen()
+
+        self.debug("Updating service offering with ID: %s" %
+                   self.service_offering_1.id)
+
+        cmd = updateServiceOffering.updateServiceOfferingCmd()
```

Comment:
Added updatedServiceOffering method to base.py , and refactored the script accordingly.